include/vkwrap/vkwrap/image.h

```diff
@@ -0,0 +1,302 @@
+#pragma once
+
+#include "common/vulkan_include.h"
+
+#include "vkwrap/core.h"
+#include "vkwrap/mman.h"
+#include "vkwrap/queues.h"
+#include "vkwrap/utils.h"
+
+#include "utils/patchable.h"
+
+#include <range/v3/range/concepts.hpp>
+#include <range/v3/range/conversion.hpp>
+
+#include <exception>
+#include <functional>
+#include <optional>
+#include <utility>
+
+#include <cassert>
+
+namespace vkwrap
+{
+
+namespace detail
+{
+
+inline vk::ImageViewType
+chooseImageViewType( vk::ImageType type )
+{
+    using vk::ImageType;
+    using vk::ImageViewType;
+
+    switch ( type )
+    {
+    case ImageType::e1D:
+        return ImageViewType::e1D;
+
+    case ImageType::e2D:
+        return ImageViewType::e2D;
+
+    case ImageType::e3D:
+        return ImageViewType::e3D;
+
+    default:
+        // Assert addresses to possible additions in VkImageType.
+        assert( 0 && "chooseImageViewType: image type is not supported." );
+        std::terminate();
+    }
+} // choosesImageViewType
+
+} // namespace detail
+
+class Image : private vk::Image
+{
+
+  private:
+    using Base = vk::Image;
+
+  private:
+    Mman* m_mman{ nullptr };
+    vk::UniqueImageView m_view;
+
+    // clang-format off
+    static constexpr vk::ImageViewCreateInfo k_view_initial_create_info{
+        // We don't use these specific fields.
+        .pNext = {},
+        .flags = {},
+
+        // Don't remap color components.
+        .components = {},
+
+        // We use images with only one mipmap and one layer.
+        .subresourceRange = { 
+            .baseMipLevel = 0, 
+            .levelCount = 1, 
+            .baseArrayLayer = 0, 
+            .layerCount = 1 } 
```

Comment:
We cannot set `layerCount` in `subresourseRange` to 1, because we use array of textures, please add functionality to setting this field